src/main/java/org/opentripplanner/graph_builder/module/islandpruning/PruneIslands.java

```diff
@@ -340,6 +338,14 @@ private void collectNeighbourVertices(
           continue;
         }
         State[] states = e.traverse(s0);
+        if (states.length > 1) {
+          throw new IllegalStateException(
+            "Expected single state returned from edge %s but received %s".formatted(
+                e,
+                states.length
+              )
+          );
+        }
```

Comment:
Here multiple states is ok, we should add all all returned vertices to neighbors.

Reply:
I did this now.